core/storage/activity/gae_models.py

```diff
@@ -39,6 +39,19 @@ def get_deletion_policy():
         """Activity references are not related to users."""
         return base_models.DELETION_POLICY.NOT_APPLICABLE
 
+    @staticmethod
```

Comment:
I think that is okay to keep it this way if this was a class method we would have unused `cls` variable and I don't see any big benefit behind this.

Reply:
I think @DubeySandeep is correct here, actually -- has_reference_to_user_id should probably be defined the same way in all models, for consistency and to avoid developer confusion (unless, @vojtechjelinek, you can point to a reference that suggests that this is a standard pattern in Python).

@vojtechjelinek, please could you fix this in a follow-up PR? Using unused_cls is fine IMO.

Reply:
@seanlip I'm okay with this, but now we have the benefit of using `@staticmethod` that we can use `assertStaticMethod` and if anyone changes this method to include some user id and model operation they need to change it to `@classmethod` and the tests will fail. 

If this was a `@classmethod` we would have to maybe create some more complicated test that would check if there is `unused_cls` and not `cls`.

Reply:
Oh, I see -- you are using "static method" as a proxy for "does not make use of user ID"? That's an interesting approach, but I don't think it would be clear to developers.

I'm not sure what the benefit behind the test you mention is -- can you explain? I understand a test mandating that all classes define a has_reference_to_user_id() method, but I'm not sure I understand what the one you mention is guarding against -- can you describe a scenario that it would catch?

Reply:
I just realized that the use case I thought about doesn't work :D, I will set all the static methods to class methods.
I just realized that the use case I thought about doesn't work :D, I will set all the static methods to class methods.

Reply:
>  can you describe a scenario that it would catch?

If we use classmethod:
We are going to write a test for this method which will pass even in the future someone changes the method, right?

But if we make sure that this method is static and adds a test for the same then if in future someone changes this method such that it makes a query using cls then the test will fail (basically this will open us to write a test which will guard the changes in this method.)

But I don't think that's more beneficial compared to the consistent behavior/property of a method for all the subclass of a parent class.

Reply:
Also, I see any model with `DELETION_POLICY.NOT_APPLICABLE` has a static method for `has_reference_to_user_id`, so should we only define this method in the subclass if the deletation policy is applicable? (I'm not sure about this idea as I'm not completely aware of how user_id reference and deletion_policiy is connected!)

Reply:
Yes, I'm not sure about the utility of the test, and I think that it is better to aim for consistency here.

Reply:
> Also, I see any model with DELETION_POLICY.NOT_APPLICABLE has a static method for has_reference_to_user_id, so should we only define this method in the subclass if the deletation policy is applicable? (I'm not sure about this idea as I'm not completely aware of how user_id reference and deletion_policiy is connected!)

This seems reasonable to me, actually. @vojtechjelinek what do you think?

Reply:
I was going for consistency and testability here with implementing `has_reference_to_user_id` even for models with `DELETION_POLICY.NOT_APPLICABLE`, but I can probably modify the tests to check if `DELETION_POLICY != NOT_APPLICABLE` when looking for `has_reference_to_user_id`.

Reply:
Yep, that was what I was thinking. Then for other classes has_reference_to_user_id will raise NotImplementedError (inherited from base class), which seems appropriate.